### DISCUSSION TITLE
Hope to godot support browser components, and also support them in the old version

### DISCUSSION THREAD

**Discussion by tinysouth** [godotengine/godot-proposals#10375] [General Discussions]
### Describe the project you are working on

Hope to godot support browser components, and also support them in the old version

### Describe the problem or limitation you are having in your project

Hope to godot support browser components, and also support them in the old version

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Hope to godot support browser components, and also support them in the old version

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Hope to godot support browser components, and also support them in the old version

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Hope to godot support browser components, and also support them in the old version

### Is there a reason why this should be core and not an add-on in the asset library?

Hope to godot support browser components, and also support them in the old version

**Comment by AThousandShips** [comment]
Converted to a discussion as the form was not filled in properly and it lacks details

**Comment by Calinou** [comment]
Are you referring to [Web Components](https://developer.mozilla.org/en-US/docs/Web/API/Web_components)? I'm not sure what they would bring in the context of Godot's web export.

**Comment by tinysouth** [comment]
yes

**Comment by tinysouth** [comment]
also hope godot can support php

**Comment by AThousandShips** [comment]
What would this be used for with Godot? Please provide some examples of what this would be used for and what problems it would solve

**Reply by tinysouth** [reply]
For example, when the learning cost of the constructor is too high,Developers can use the knowledge they have learned about the web html development programming language to develop relevant godot ui components instead of gd language.

**Reply by tinysouth** [reply]
> What would this be used for with Godot? Please provide some examples of what this would be used for and what problems it would solve

I don't have enough time and energy to modify godot source code and then add a webview component for godot.I was going to try this before.With the addition of the webview component, we can create more polished godot ui.

**Reply by AThousandShips** [reply]
There isn't any features or functionality that isn't covered by Godot already? Just a separate system? This wouldn't allow you to solve things that Godot doesn't already if you just learn the Godot UI components?

**Reply by Calinou** [reply]
[Adding a built-in WebView](https://github.com/godotengine/godot-proposals/issues/960) was discussed in a past proposal, and it was rejected.

However, nothing prevents a WebView from being implemented by a third-party add-on.

**Reply by tinysouth** [reply]
> [Adding a built-in WebView](https://github.com/godotengine/godot-proposals/issues/960) was discussed in a past proposal, and it was rejected.
> 
> However, nothing prevents a WebView from being implemented by a third-party add-on.

but third-party add-on is not easy to crossplatform
